iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp

```
@@ -209,6 +244,307 @@ class GlobalStoreOpConversion : public OpConversionPattern<StoreOpTy> {
+    // iree_vm_list_create(&element_type, {initial_capacity}, state->allocator, &list);
+    // iree_vm_ref_wrap_assign(list, iree_vm_list_type_id(), &list_ref));
+    // clang-format on
+
+    auto ctx = allocOp.getContext();
+    auto loc = allocOp.getLoc();
+
+    auto listType = allocOp.getType()
+                        .cast<IREE::VM::RefType>()
+                        .getObjectType()
+                        .cast<IREE::VM::ListType>();
+    auto elementType = listType.getElementType();
+    std::string elementTypeStr;
+    StringRef elementTypeConstructor;
+    if (elementType.isa<IntegerType>()) {
+      unsigned int bitWidth = elementType.getIntOrFloatBitWidth();
+      elementTypeStr =
+          std::string("IREE_VM_VALUE_TYPE_I") + std::to_string(bitWidth);
+      elementTypeConstructor = "iree_vm_type_def_make_value_type";
+    } else {
+      return allocOp.emitError() << "Unhandeled element type " << elementType;
+    }
+
+    auto elementTypeOp = rewriter.create<emitc::CallOp>(
+        /*location=*/loc,
+        /*type=*/emitc::OpaqueType::get(ctx, "iree_vm_type_def_t"),
+        /*callee=*/rewriter.getStringAttr(elementTypeConstructor),
+        /*args=*/ArrayAttr::get(ctx, {StringAttr::get(ctx, elementTypeStr)}),
+        /*templateArgs=*/ArrayAttr{},
+        /*operands=*/ArrayRef<Value>{});
+
+    auto elementTypePtrOp = rewriter.create<emitc::GetAddressOfOp>(
+        /*location=*/loc,
+        /*result=*/emitc::OpaqueType::get(ctx, "iree_vm_type_def_t*"),
+        /*operand=*/elementTypeOp.getResult(0));
+
+    auto listOp = rewriter.create<emitc::ConstOp>(
+        /*location=*/loc,
+        /*resultType=*/emitc::OpaqueType::get(ctx, "iree_vm_list_t*"),
+        /*value=*/StringAttr::get(ctx, "NULL"));
+
+    auto listPtrOp = rewriter.create<emitc::GetAddressOfOp>(
+        /*location=*/loc,
+        /*result=*/emitc::OpaqueType::get(ctx, "iree_vm_list_t**"),
+        /*operand=*/listOp.getResult());
+
+    rewriter.create<emitc::CallOp>(
```

These calls can fail - anything with iree_status_t you'll want to insert an IREE_RETURN_IF_ERROR around and propagate the error back up. This is especially important with calls to external modules as for example the HAL failing to load an executable at runtime may fail and if you ignore the error things will get crashy fast.

---

I've added these, or rather something similar as we can't emit a macro invocation with a function call as argument. So the newly added test at least fails as expected.
 
We are leaking memory though as we don't release the refs before returning. Let's discuss directly how to properly support this.